Cannot process `number` fields with format `decimal`

The new API version for `Consumption` `2023-03-01` has fields containing decimals which result in an error when trying to parse the swagger.

Swagger:
```
    "LegacyReservationTransactionProperties": {
      "description": "The properties of a legacy reservation transaction.",
      "properties": {
      ...
        "quantity": {
          "description": "The quantity of the transaction.",
          "type": "number",
          "format": "decimal",
          "readOnly": true
        },
        "amount": {
          "description": "The charge of the transaction.",
          "type": "number",
          "format": "decimal",
          "readOnly": true
        },
      ...
```

Error:
```
Error: parsing Swagger files: parsing files in "../../../swagger/specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-03-01": parsing definition: finding resources for tag "ReservationTransactions": finding operations: parsing GET operation "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/reservationTransactions": building options for operation "List": determining field type for operation: unsupported format type for number "decimal"
    run_validate_can_parse_data_test.go:75: error: parsing Swagger files: parsing files in "../../../swagger/specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-03-01": parsing definition: finding resources for tag "ReservationTransactions": finding operations: parsing GET operation "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.Consumption/reservationTransactions": building options for operation "List": determining field type for operation: unsupported format type for number "decimal"
```

We need to update the parser within `importer-rest-specs-api` to handle fields in both the options (header/query strings) as well as the request body to map decimals to floats in the data API.

--- a/tools/importer-rest-api-specs/components/parser/operations.go
+++ b/tools/importer-rest-api-specs/components/parser/operations.go
@@ -205,6 +205,12 @@ func (p operationsParser) determineObjectDefinitionForOptionRaw(paramType string
 				}, nil
 			}
 
+			if strings.EqualFold(format, "decimal") {
+				return &models.ObjectDefinition{
+					Type: models.ObjectDefinitionFloat,
+				}, nil
+			}
+
 			if format != "" {
 				// we don't know what this is, better to raise an error and handle it than make
 				// it an integer if it should be a float or something

--- a/tools/importer-rest-api-specs/components/parser/operations_test.go
+++ b/tools/importer-rest-api-specs/components/parser/operations_test.go
@@ -2770,8 +2770,8 @@ func TestParseOperationSingleWithHeaderOptions(t *testing.T) {
 		t.Fatal("expected a non-long running operation but it was long running")
 	}
 
-	if len(world.Options) != 6 {
-		t.Fatalf("expected HeadWorld to have 6 options but got %d", len(world.Options))
+	if len(world.Options) != 7 {
+		t.Fatalf("expected HeadWorld to have 7 options but got %d", len(world.Options))
 	}
 
 	boolOption, ok := world.Options["BoolValue"]
@@ -2829,6 +2829,23 @@ func TestParseOperationSingleWithHeaderOptions(t *testing.T) {
 	}
 	if csvOfStringOption.HeaderName != nil && *csvOfStringOption.HeaderName != "csvOfStringValue" {
 		t.Fatalf("expected HeadWorld Option 'CsvOfStringValue's HeaderName to be `csvOfStringValue` but got %q", *csvOfStringOption.HeaderName)
+	}
+
+	decimalOption, ok := world.Options["DecimalValue"]
+	if !ok {
+		t.Fatalf("expected HeadWorld Options to contain 'DecimalValue' but didn't get it")
+	}
+	if decimalOption.ObjectDefinition.Type != models.ObjectDefinitionFloat {
+		t.Fatalf("expected HeadWorld Option 'DecimalValue' to be a Float but got %q", string(decimalOption.ObjectDefinition.Type))
+	}
+	if decimalOption.QueryStringName != nil {
+		t.Fatalf("expected HeadWorld Option 'DecimalValue's QueryStringName to be nil but got %q", *decimalOption.QueryStringName)
+	}
+	if decimalOption.HeaderName == nil {
+		t.Fatalf("expected HeadWorld Option 'DecimalValue's HeaderName to be `decimalValue` but was nil")
+	}
+	if decimalOption.HeaderName != nil && *decimalOption.HeaderName != "decimalValue" {
+		t.Fatalf("expected HeadWorld Option 'DecimalValue's HeaderName to be `decimalValue` but got %q", *decimalOption.HeaderName)
 	}
 
 	doubleOption, ok := world.Options["DoubleValue"]

--- a/tools/importer-rest-api-specs/components/parser/testdata/operations_single_with_header_options.json
+++ b/tools/importer-rest-api-specs/components/parser/testdata/operations_single_with_header_options.json
@@ -59,6 +59,14 @@
             }
           },
           {
+            "name": "decimalValue",
+            "in": "header",
+            "required": true,
+            "type": "number",
+            "format": "decimal",
+            "description": "Some Decimal Value which should be output as Float64"
+          },
+          {
             "name": "doubleValue",
             "in": "header",
             "required": true,

